app/lib/pages/settings/changelog/change.dart

```diff
@@ -85,14 +118,21 @@ class Version {
     return name.hashCode;
   }
 
-  int _versionNameToInt(String version) {
-    if (version == null) return -1;
-    assert(versionRegEx.allMatches(version).length == 1,
-        "The version string should be in a valid format");
-    final List<String> splittedVersion = version.split('.');
-    final String versionWithoutPoints =
-        '${splittedVersion[0]}${splittedVersion[1]}${splittedVersion[2]}';
-    final int versionAsInt = int.parse(versionWithoutPoints);
-    return versionAsInt;
+  /// Returns the difference between two versions.
+  ///
+  /// If the first version is newer than the second one, the result is positive.
+  /// If the first version is older than the second one, the result is negative.
+  /// If the versions are equal, the result is 0.
+  int _compare(Version a, Version b) {
+    if (a.major > b.major) return 1;
+    if (a.major < b.major) return -1;
+
+    if (a.minor > b.minor) return 1;
+    if (a.minor < b.minor) return -1;
+
+    if (a.patch > b.patch) return 1;
+    if (a.patch < b.patch) return -1;
+
+    return 0;
```

Comment:
Btw with `package:collection` you could write the code like this:

```suggestion
  int _compare(Version a, Version b) =>
      _compareMajor.then(_compareMinor).then(_comparePatch).call(a, b);

  int _compareMajor(Version a, Version b) => a.major.compareTo(b.major);
  int _compareMinor(Version a, Version b) => a.minor.compareTo(b.minor);
  int _comparePatch(Version a, Version b) => a.patch.compareTo(b.patch);
```

For me both ways are fine, just wanted to show you :)

Reply:
Somehow this doesn't compile for me. With ChatGPT as a my second code review partner it found another smart way: Using the *Comparble` interface:

```dart
class Version implements Comparable<Version> {

  ...

  /// Compares this version to another [Version].
  ///
  /// Returns a negative value if this version is older than [other], a positive
  /// value if it's newer, and 0 if they're identical.
  @override
  int compareTo(Version other) {
    if (major != other.major) return major.compareTo(other.major);
    if (minor != other.minor) return minor.compareTo(other.minor);
    return patch.compareTo(other.patch);
  }
```